server/src/test/java/org/elasticsearch/action/search/SearchAsyncActionTests.java

```diff
@@ -263,26 +257,17 @@ public void testFanOutAndCollect() throws InterruptedException {
         request.allowPartialSearchResults(true);
         request.setMaxConcurrentShardRequests(randomIntBetween(1, 100));
         AtomicReference<TestSearchResponse> response = new AtomicReference<>();
-        ActionListener<SearchResponse> responseListener = new ActionListener<SearchResponse>() {
-            @Override
-            public void onResponse(SearchResponse searchResponse) {
-                response.set((TestSearchResponse) searchResponse);
-            }
-
-            @Override
-            public void onFailure(Exception e) {
-                logger.warn("test failed", e);
-                fail(e.getMessage());
-            }
-        };
+        ActionListener<SearchResponse> responseListener = ActionListener.wrap(
+            searchResponse -> response.set((TestSearchResponse) searchResponse), e -> fail(e.getMessage()));
```

Comment:
`fail(e.getMessage())` loses the stack trace from the thrown exception, which might be useful. How about `throw new AssertionError("unexpected", e);` instead?